### ISSUE TITLE
Using `Vec<f64>` as parameters only works when adding `argmin-math` as dependency

### ISSUE THREAD

**Issue by StefanMathis** [argmin-rs/argmin#306]
In a downstream crate, I added `argmin 0.7.0` as a dependency (but not `argmin-math`). I then tried to setup a solver like this:

```rust
let solver = NelderMead::new(vec![
    vec![3.0f64, 3.0f64],
    vec![2.0f64, 1.5f64],
    vec![1.0f64, 0.5f64],
]).unwrap();
```
This fails with the error message that the trait `ArgminAdd` is not implemented for `Vec<f64>`. However, once I add `argmin-math` as a direct dependency of the downstream crate as well, it works (it is not necessary to import the trait).

Is this expected behaviour?

**Comment by stefan-k** [comment]
Yes, this is intended and documented in the [book](http://www.argmin-rs.org/book/using_argmin.html). Unfortunately there are currently some difficulties to actually find the book, depending on which path you took ;) I assume that your path was either via the API documentation or the examples, which both so far make no mention of the book. The former is fixed in #270 (but will only go live with the next release) and I will fix the latter by turning the examples into their own crates with dedicated `Cargo.toml` files hopefully soon. 

It may seem a bit tedious having to add both `argmin` and `argmin-math` but this split approach has a couple of advantages, in particular in terms of maintainability and in terms of flexibility for the users. For instance, it allows one to cut down on compilation time and code size if needed by enabling one to explicitly specify which functionality is needed. 

Apologies for the inconvenience and thanks for this issue, because it really helps improving the "docs experience".

**Comment by StefanMathis** [comment]
Thank you for your fast and detailed answer. Your assumptions were correct, I went the path of API documentation + examples.

I think the split approach is the right one, keep up the good work :-) I consider this issue to be resolved.
